core-primitives/time-utils/src/lib.rs

```diff
@@ -60,10 +63,19 @@ mod tests {
 	use super::*;
 
 	#[test]
-	fn subsequent_nows_are_increasing_in_time() {
+	fn subsequent_nows_with_sleep_inbetween_are_strictly_increasing_in_time() {
 		let before = duration_now();
+		thread::sleep(Duration::from_nanos(1));
```

Comment:
After reading the documentation of [`SystemTime`](https://doc.rust-lang.org/std/time/struct.SystemTime.html), with the paragraph
```
In other words, an operation that happens after another operation in real time may have an earlier SystemTime!
```
I'm not so sure anymore if we'll ever get a stable test, even with the sleep (maybe if we sleep long enough, like a second or so). The resolution of `SystemTime` seems to vary, depending on your OS, but it seems to be in the range of nanoseconds to 100s of nanoseconds. So I would increase the sleep to at least a millisecond for now.